src/code42cli/cmds/search/options.py

```diff
@@ -174,10 +175,37 @@ def create_search_options(search_term):
     )
 
     def search_options(f):
-
         f = search_interval_options(f, search_term)
         f = checkpoint_option(f)
         f = advanced_query_option(f)
         return f
 
     return search_options
+
+
+def server_options(f):
+    hostname_arg = click.argument("hostname")
+    protocol_option = click.option(
+        "-p",
+        "--protocol",
+        type=click.Choice(ServerProtocol(), case_sensitive=False),
+        default=ServerProtocol.UDP,
+        help="Protocol used to send logs to server. "
+        "Use TLS for additional security. Defaults to UDP.",
```

Comment:
what about `--secure` option which accepts the `certs` file, i.e `--secure <cert file>`.  
Let the protocol's options remain as TCP or UDP as earlier. 

With current options, we have to cover cases like what if user passes `-p  TCP --certs <cert file>` or `-p UDP --certs <cert file>`? Do we make secure data transmission? 
Having `-p TCP --secure <cert file>` makes intent more clear where one is trying to say "I wan to transfer the data more securely" and `--secure` is not passed they are ok with insecure data transmission.

We can explain that we use TLS encryption in `--secure` help message.

Reply:
Yeay maybe. But there are caveats there similarly.

Like protocol has to always be `TCP` when using secure. There is no secure `UDP`.

Reply:
Plus `--secure` would have to both work as a flag and as an option because you can use NoCert option for TLS

Reply:
Wondering if there will be a use case when a user would prefer insecure TCP communication over secure.

Reply:
unlike http/s, udp/tcp log transmission is still very commonly done insecurely since it typically is all done from within the same network. This is why we haven't even added this feature until now.

insecure should remain the default because it is still more common and also because changing this would be a breaking change.

Reply:
I honestly can go either way on the design of this, `--protocol TCP --secure <optional-certs-files>` or `--protocol TLS --certs <certs-files>` or another way.

It does seem odd to have to explicitly say I want `--protocol TCP` if you want to use TLS-TCP.

But having the certs option only apply to one protocol is also a drawback

Reply:
Looking at all the different scenarios and given that UDP still remains insecure, what about 

-  Defaults to UDP when protocol is not specified, as it is now.
- --protocol UDP     # Same as above just that the user is being explicit 
-  --protocol TCP     # Default insecure when TCP is specified
- --protocol TCP --TLS     # Secure TCP, --TLS should be allowed only when TCP is specified
- --protocol TCP --CA <CERT> # secure TCP, --CA should be allowed only when TCP is specified

Reply:
I would say that `--certs` could default to using TCP, and error if you specify a protocol known to be not compatible (in this case UDP).

I would prefer `--certs` over `--secure` since passing certs locations  to a param named `secure` makes a little less sense contextually, imo.

Reply:
OK, but take note to all of the complexitie that arise from doing it this way ( in the next commit)

Reply:
For example, this is what the new help text for protocol will be like: 

```
        help="Protocol used to send logs to server. "
        "Defaults to UDP unless providing certs, than it defaults to TCP.",
```

Reply:
Also, does this mean we are not going to support `ssl.CERT_NONE` ?

Reply:
If we do want to support no certs, how would one do that?

`send-to --certs None`

That command does really seem like it is using TLS, yet it can be....

Reply:
Reasons like this (And more!) are why I thought it would make more sense to have TLS be part of the protocol... So, here is another middle-ground:


`code42 security-data send-to --protocol TLS-TCP`

Add certs optionally.

Reply:
You make a valid point about the default protocol being oddly conditional, which I agree should be avoided.
`CERT_NONE` is effectively just a way to bypass certificate validation, which is probably okay to support here for those that _really_ want it since this feature is mostly targeted at on-prem services that could have self-signed certs.